regression/polus-bleed-through-estimation-plugin/src/bench.py

```diff
@@ -0,0 +1,269 @@
+            coefficients_after[i, j] = coefficients_before[j, i] = after
+
+        with open(before_path, 'wb') as outfile:
+            pickle.dump(coefficients_before, outfile)
+        with open(after_path, 'wb') as outfile:
+            pickle.dump(coefficients_after, outfile)
+
+    else:
+        with open(before_path, 'rb') as infile:
+            coefficients_before = pickle.load(infile)
+        with open(after_path, 'rb') as infile:
+            coefficients_after = pickle.load(infile)
+
+    draw_heatmap(coefficients_before, coefficients_after, heatmaps_dir, heatmaps_name, replicate)
+
+    return
+
+
+def draw_heatmap(
+        before: numpy.ndarray,
+        after: numpy.ndarray,
+        heatmaps_dir: Path,
+        heatmaps_name: str,
+        replicate: int,
+):
+    heat = numpy.zeros_like(before, dtype=numpy.float32)
+    heat[numpy.tril_indices_from(heat)] = after[numpy.tril_indices_from(after)]
+    heat[numpy.triu_indices_from(heat)] = before[numpy.triu_indices_from(before)]
+    heat = heat.T
+
+    mask = numpy.zeros_like(heat, dtype=bool)
+    mask[numpy.diag_indices_from(mask)] = True
+
+    with seaborn.axes_style('white'):
+        _ = seaborn.heatmap(
+            heat, mask=mask, vmin=-1, vmax=1, cmap='vlag', linewidth=0.5,
+            annot=True, fmt='.3f', annot_kws={'size': 8},
+        )
+        matplotlib.pylab.title(f'Replicate {replicate}')
+
+        plot_path = heatmaps_dir.joinpath(f'{heatmaps_name}.png')
+        matplotlib.pylab.savefig(str(plot_path), dpi=300)
+
+    matplotlib.pylab.close('all')
+
+    return
+
+
+def bench(file_pattern: str, replicate: int):
+    data_dir = Path('../../../data/bleed_through_estimation').resolve()
+    assert data_dir.exists()
+
+    images_dir = data_dir.joinpath('rat_brain_input')
+    assert images_dir.exists()
+
+    components_dir = data_dir.joinpath('rat_brain_output').joinpath('images')
+    assert components_dir.exists()
+
+    coefficients_dir = data_dir.joinpath('rat_brain_output').joinpath('csvs')
+    assert coefficients_dir.exists()
+
+    corrected_dir = data_dir.joinpath('rat_brain_corrected')
+    assert corrected_dir.exists()
+
+    heatmaps_dir = data_dir.joinpath('plots')
+    assert heatmaps_dir.exists()
+
+    channel_ordering = [1, 0, 3, 2, 4, 5, 7, 6, 8, 9]
+    fp = FilePattern(images_dir, file_pattern)
+    for group in fp(['c']):
+        write_corrected_images(
+            group=group,
+            channel_ordering=channel_ordering,
+            components_dir=components_dir,
+            output_dir=corrected_dir,
+        )
+        plot_pearson_coefficients(
+            group=group,
+            pattern=file_pattern,
+            channel_ordering=channel_ordering,
+            corrected_dir=corrected_dir,
+            selector_name='HighMeanIntensity',
+            heatmaps_dir=heatmaps_dir,
+            replicate=replicate,
+        )
+    return
+
+
+if __name__ == '__main__':
```

Comment:
testing artifact?

Reply:
I wanted this as a script that a reader can run to recreate the benchmarks in the paper.